src/openMVG/matching_image_collection/gpu/LatchBitMatcher.cpp

```diff
@@ -0,0 +1,144 @@
+  memset(&resDescQuery, 0, sizeof(resDescQuery));
+  resDescQuery.resType = cudaResourceTypeLinear;
+  resDescQuery.res.linear.devPtr = m_dQuery;
+  resDescQuery.res.linear.desc.f = cudaChannelFormatKindUnsigned;
+  resDescQuery.res.linear.desc.x = 32;
+  resDescQuery.res.linear.desc.y = 32;
+  resDescQuery.res.linear.sizeInBytes = sizeD;
+  
+  struct cudaTextureDesc texDescQuery;
+  memset(&texDescQuery, 0, sizeof(texDescQuery));
+  texDescQuery.addressMode[0] = cudaAddressModeBorder;
+  texDescQuery.addressMode[1] = cudaAddressModeBorder;
+  texDescQuery.filterMode = cudaFilterModePoint;
+  texDescQuery.readMode = cudaReadModeElementType;
+  texDescQuery.normalizedCoords = 0;
+  
+  struct cudaResourceDesc resDescTraining;
+  memset(&resDescTraining, 0, sizeof(resDescTraining));
+  resDescTraining.resType = cudaResourceTypeLinear;
+  resDescTraining.res.linear.devPtr = m_dTraining;
+  resDescTraining.res.linear.desc.f = cudaChannelFormatKindUnsigned;
+  resDescTraining.res.linear.desc.x = 32;
+  resDescTraining.res.linear.desc.y = 32;
+  resDescTraining.res.linear.sizeInBytes = sizeD;
+  
+  struct cudaTextureDesc texDescTraining;
+  memset(&texDescTraining, 0, sizeof(texDescTraining));
+  texDescTraining.addressMode[0] = cudaAddressModeBorder;
+  texDescTraining.addressMode[1] = cudaAddressModeBorder;
+  texDescTraining.filterMode = cudaFilterModePoint;
+  texDescTraining.readMode = cudaReadModeElementType;
+  texDescTraining.normalizedCoords = 0;
+
+  m_texQuery = 0;
+  m_texTraining = 0;
+
+  checkError(cudaCreateTextureObject(&m_texQuery, &resDescQuery, &texDescQuery, nullptr));
+  checkError(cudaCreateTextureObject(&m_texTraining, &resDescTraining, &texDescTraining, nullptr));
+}
+
+void LatchBitMatcher::match(void* h_descriptorsQuery, void* h_descriptorsTraining, int numKPQuery, int numKPTraining) {
+  m_numKPQuery = numKPQuery;
+  m_numKPTraining = numKPTraining;
+
+  const size_t sizeDQuery = numKPQuery * 64; // D1 for descriptor1
+  const size_t sizeDTraining = numKPTraining * 64; // D2 for descriptor2
+  const size_t sizeMatches = m_maxKP * sizeof(int); // M for Matches
+
+  cudaMemsetAsync(m_dQuery, 0, sizeDQuery, m_stream1);
+  cudaMemsetAsync(m_dTraining, 0, sizeDTraining, m_stream2);
+
+  cudaMemsetAsync(m_dMatches1, 0, sizeMatches, m_stream1);
+  cudaMemsetAsync(m_dMatches2, 0, sizeMatches, m_stream2);
+
+  cudaMemcpyAsync(m_dQuery, h_descriptorsQuery, sizeDQuery, cudaMemcpyHostToDevice, m_stream1);
+  cudaMemcpyAsync(m_dTraining, h_descriptorsTraining, sizeDTraining, cudaMemcpyHostToDevice, m_stream2);
+ 
+  cudaStreamSynchronize(m_stream1);
+  cudaStreamSynchronize(m_stream2);
+
+  cudaBruteForceMatcher(m_dTraining, m_numKPTraining, m_texQuery, m_numKPQuery, m_dMatches1, m_stream1);
+  cudaBruteForceMatcher(m_dQuery, m_numKPQuery, m_texTraining, m_numKPTraining, m_dMatches2, m_stream2);
+}
+
+openMVG::matching::IndMatches LatchBitMatcher::retrieveMatches(float ratio) {
+  int h_Matches1[m_maxKP];
```

Comment:
If the size `m_maxKP` is const I think it can be ok.
Seems like it is const -> https://github.com/openMVG/openMVG/pull/935/files#diff-97b83af8e48e7a4a8e91dbb47c926ba2R28

So I will check later on.